src/app.js

```diff
@@ -91,48 +90,220 @@ app.routeAsync("/charon/*")
+  .getAsync(ifDatasetExists, sendAuspiceEntrypoint)
+;
+
+
+/* Users
+ */
+app.routeAsync("/whoami")
+  .getAsync(users.getWhoami);
+
+/* For requests for text/html, the first /whoami implementation returned bare
+ * bones HTML dynamic on the logged in user.  This was later replaced by a
+ * redirect to /users/:name (for a logged in session) or /users (if no one
+ * logged in), which then rendered a Gatsby page.  However, the /users/:name
+ * form was purely aesthetic: :name was not used for routing at all.  Instead
+ * of keeping up this fiction, have /whoami render the Gatsby page directly,
+ * at least until we _actually_ implement user profile pages.
+ *   -trs, 4 Oct 2021
+ */
+app.route(["/users", "/users/:name"])
+  .get((req, res) => res.redirect("/whoami"));
+
+
+/* Auspice HTML pages and assets.
+ *
+ * Auspice hardcodes URL paths that start with /dist/… in its Webpack config,
+ * so we must use that prefix here too.
+ */
+app.route("/dist/*")
+  .all(expressStaticGzip(auspiceAssetPath(), {fallthrough: false, maxAge: '30d'}));
+
+
+/* Gatsby static HTML pages and other assets.
+ *
+ * Any URLs matching Gatsby's static files will be handled here, e.g.
+ * static-site/public/influenza/index.html will be served for /influenza.
+ *
+ * When a Gatsby dev server is in use, assets are proxied from the dev server
+ * instead of served straight from disk.
+ */
+if (app.locals.gatsbyDevUrl) {
+  // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+  const {createProxyMiddleware} = require("http-proxy-middleware");
+
+  // WebSocket endpoint
+  app.get("/socket.io/", createProxyMiddleware(app.locals.gatsbyDevUrl, {ws: true}));
+
+  /* This is the end of the line in gatsbyDevUrl mode, as the proxy middleware
+   * doesn't fallthrough on 404 and even if it did the Gatsby dev server
+   * returns a 200 Ok for missing pages.
+   */
+  app.use(createProxyMiddleware(app.locals.gatsbyDevUrl));
+} else {
+  app.use(express.static(gatsbyAssetPath()));
 }
 
+
+/* Everything else gets 404ed.
+ */
+app.use((req, res, next) => next(new NotFound()));
+
+
+/* Error handler
+ */
+app.useAsync(async (err, req, res, next) => {
+  /* XXX TODO: Replace calls to Express' next() with calls to our own custom
+   * finalhandler (the library Express uses) function configured with an
+   * "onerror" handler that does what we want with regard to logging.
+   *   -trs, 1 Oct 2021
+   */
+  if (res.headersSent) {
+    utils.verbose("Headers already sent; using Express' default error handler");
+    return next(err);
+  }
+
+  if (err instanceof NotFound) {
+    /* A note about routing: if the current URL path (i.e. req.path) matches a
+     * a page known to Gatsby (e.g. via the Gatsby page's "path" or "matchPath"
+     * properties), then Gatsby will perform client-side routing to load that
+     * page even though we're serving a static page (404.html) here.
+     */
+    utils.verbose(`Sending Gatsby 404`);
```

Comment:
The verbose logging is perhaps a little too verbose - this statement will end up followed by another in `sendGatsbyPage`. I'd think that this could be removed.

Reply:
Sure, done.

Relatedly, I think it'd be great to adopt a more structured logging framework so we can include more logging in the code and tune output via config/environ instead of adjusting code. It's impossible to strike the right balance for all situations, so defaulting to "log it" and having knobs is key.